go-controller/pkg/controller/controller.go

```diff
@@ -80,7 +81,7 @@ func (c *controller[T]) Start(threadiness int) error {
 		return fmt.Errorf("failed to add event handler: %w", err)
 	}
 
-	if !util.WaitForNamedCacheSyncWithTimeout(c.name, c.stopChan, c.eventHandler.HasSynced) {
+	if !util.WaitForHandlerSyncWithTimeout(c.name, c.stopChan, 20*time.Second, c.eventHandler.HasSynced) {
```

Comment:
oh I meant to change this to informer sync, so use `c.config.Informer.HasSynced` instead of `c.eventHandler.HasSynced`

Reply:
Oh I see, I read your other message too quickly.

-  in services_controller.go (https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/controller/services/services_controller.go#L198) I definitely need `WaitForHandlerSyncWithTimeout`, because I want all service keys to be added to the queue at that step, since I removed that part from the node tracker startup.

- here in controller.go the existing code passes the handler `HasSync`, so I simply figured I'd keep the existing logic and use `WaitForHandlerSyncWithTimeout`. If that's not necessary, I can definitely change that.